core/src/test/java/com/linecorp/armeria/server/logging/AnnotatedServiceLogNameTest.java

```diff
@@ -0,0 +1,177 @@
+        client.get("/default/service-name").aggregate().join();
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx))
+                .isEqualTo(MyAnnotatedService.class.getName());
+        assertThat(sctx.config().defaultServiceName()).isEqualTo(MyAnnotatedService.class.getName());
+        assertThat(sctx.log().whenComplete().join().serviceName())
+                .isEqualTo(MyAnnotatedService.class.getName());
+    }
+
+    @Test
+    void serviceName_withDefaultNaming() {
+        client.get("/default-naming/service-name").aggregate().join();
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx)).isEqualTo("DefaultNaming");
+        // ServiceNaming is used.
+        assertThat(sctx.config().defaultServiceName()).isNull();
+        assertThat(sctx.log().whenComplete().join().serviceName()).isEqualTo("DefaultNaming");
+    }
+
+    @Test
+    void serviceName_withSimpleNaming() {
+        client.get("/simple-naming/service-name").aggregate().join();
+        final String expectedServiceName = AnnotatedServiceLogNameTest.class.getSimpleName() + '$' +
+                                           MyAnnotatedService.class.getSimpleName();
+
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx))
+                .isEqualTo(expectedServiceName);
+        // ServiceNaming is used.
+        assertThat(sctx.config().defaultServiceName()).isNull();
+        assertThat(sctx.log().whenComplete().join().serviceName())
+                .isEqualTo(expectedServiceName);
+    }
+
+    @Test
+    void serviceName_withFullNaming() {
+        client.get("/full-naming/service-name").aggregate().join();
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx))
+                .isEqualTo(MyAnnotatedService.class.getName());
+        // ServiceNaming is used.
+        assertThat(sctx.config().defaultServiceName()).isNull();
+        assertThat(sctx.log().whenComplete().join().serviceName())
+                .isEqualTo(MyAnnotatedService.class.getName());
+    }
+
+    @Test
+    void serviceName_withAnnotation() {
+        client.get("/annotation/service-name").aggregate().join();
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx))
+                .isEqualTo("MyService");
+        assertThat(sctx.config().defaultServiceName()).isEqualTo("MyService");
+        assertThat(sctx.log().whenComplete().join().serviceName())
+                .isEqualTo("MyService");
+    }
+
+    @Test
+    void serviceName_withAnonymous() {
+        client.get("/anonymous-simple/service-name").aggregate().join();
+        assertThat(sctx.config().defaultServiceNaming().serviceName(sctx))
+                .isEqualTo("AnnotatedServiceLogNameTest$1$1");
+        // ServiceNaming is used.
+        assertThat(sctx.config().defaultServiceName()).isNull();
+        assertThat(sctx.log().whenComplete().join().serviceName())
+                .isEqualTo("AnnotatedServiceLogNameTest$1$1");
```

Comment:
I think this may fail when we add more anonymous classes to this class later. How about using some regular expression, like `^AnnotatedServiceLogNameTest(\$[0-9]+)+$` ?